src/pre_process/m_data_output.fpp

```diff
@@ -661,13 +698,23 @@ contains
                     end do
                 end if
             else
-                do i = 1, sys_size !TODO: check if this is right
-                    !            do i = 1, adv_idx%end
-                    var_MOK = int(i, MPI_OFFSET_KIND)
+                if (down_sample) then
+                    do i = 1, sys_size !TODO: check if this is right
+                        !            do i = 1, adv_idx%end
+                        var_MOK = int(i, MPI_OFFSET_KIND)
 
-                    call MPI_FILE_WRITE_ALL(ifile, MPI_IO_DATA%var(i)%sf, data_size, &
-                                            mpi_p, status, ierr)
-                end do
+                        call MPI_FILE_WRITE_ALL(ifile, q_cons_temp(i)%sf, data_size, &
+                                                mpi_p, status, ierr)
+                    end do
+                else
+                    do i = 1, sys_size !TODO: check if this is right
```

Comment:
todo: is this right?

Reply:
Yes, this is correct. I copied this code. This TODO comment has been around forever, even though the code has been correct.

Reply:
can we get rid of it if it is correct?